kyo-scheduler/jvm/src/main/scala/kyo/scheduler/Scheduler.scala

```diff
@@ -176,22 +178,25 @@ final class Scheduler(
         }
 
     private val cycleTask =
-        timerExecutor.scheduleAtFixedRate(
-            () => cycleWorkers(),
-            cycleNs,
-            cycleNs,
-            TimeUnit.NANOSECONDS
+        timerExecutor.submit(
```

Comment:
Instead of relying on Java's scheduled executor, which has additional overhead, the cycling now steals a thread and runs in a while loop with a fixed delay interval.